src/component/container/page_list.tsx

```diff
@@ -11,14 +11,111 @@ export interface PageListProps {
 	store: Store;
 }
 
+export interface PageListItemProps {
+	pageID: string;
+	name: string;
+	pageRef: PageRef;
+	store: Store;
+}
+
+@observer
+export class PageListItem extends React.Component<PageListItemProps> {
+	@MobX.observable protected pageElementEditable: boolean = false;
+	@MobX.observable
+	protected pageNameInputValue: string = this.pageNameInputValue || this.props.name;
+
+	public constructor(props: PageListItemProps) {
+		super(props);
+
+		this.handleInputChange = this.handleInputChange.bind(this);
+		this.handleBlur = this.handleBlur.bind(this);
+		this.handlePageKeyDown = this.handlePageKeyDown.bind(this);
+		this.handlePageClick = this.handlePageClick.bind(this);
+		this.handlePageDoubleClick = this.handlePageDoubleClick.bind(this);
+		this.renamePage = this.renamePage.bind(this);
+	}
+	public render(): JSX.Element {
+		return (
+			<DropdownItemEditableLink
+				editable={this.pageElementEditable}
+				focused={this.pageElementEditable}
+				handleChange={this.handleInputChange}
+				handleClick={this.handlePageClick}
+				handleDoubleClick={this.handlePageDoubleClick}
+				handleKeyDown={this.handlePageKeyDown}
+				name={this.props.name}
+				handleBlur={this.handleBlur}
+				value={this.pageNameInputValue}
+			/>
+		);
+	}
+
+	protected handlePageClick(e: React.MouseEvent<HTMLElement>): void {
+		e.preventDefault();
+		this.props.store.openPage(this.props.pageID);
+	}
+
+	@MobX.action
+	protected handleBlur(): void {
+		this.pageElementEditable = false;
+		this.pageNameInputValue = this.props.name;
+	}
+
+	@MobX.action
+	protected handlePageDoubleClick(): void {
+		this.pageElementEditable = !this.pageElementEditable;
+	}
+
+	@MobX.action
+	protected handlePageKeyDown(e: React.KeyboardEvent<HTMLInputElement>): void {
+		switch (e.key.toString()) {
+			case 'Escape':
+				this.pageNameInputValue = this.props.name;
+				this.pageElementEditable = false;
+				break;
+
+			case 'Enter':
+				if (!this.pageNameInputValue) {
+					this.pageNameInputValue = this.props.name;
+					this.pageElementEditable = false;
+					return;
+				}
+
+				this.renamePage(this.pageNameInputValue);
+				this.pageElementEditable = false;
+				this.pageNameInputValue = this.props.name;
```

Comment:
If I understand this whole feature right this line doesn't make sense as you set the name already over the 'renamePage' function. As this is asynchronous it could even get a bit dangerous.

Reply:
You're right. It should not be there my bad. Just fixed 👍